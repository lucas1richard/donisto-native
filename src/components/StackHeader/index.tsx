import * as React from 'react';
import { View, StyleSheet, SafeAreaView } from 'react-native';
import Left from './components/Left';
import Right from './components/Right';
import NavigationService from 'containers/Navigation/NavigationService';
import theme from 'theme/iftheme';
import H1 from 'components/H1';

export interface IStackHeaderProps {
  title?: string;
  noCloseOption?: boolean;
  noBackOption?: boolean;
  onClose?: () => any
}

class StackHeader extends React.Component<IStackHeaderProps> {
  static defaultProps = {
    title: ''
  };

  render() {
    const {
      noBackOption,
      noCloseOption,
      title,
      onClose
    } = this.props;
    return (
      <SafeAreaView>
        <View style={styles.wrapper}>
          <Left
            noBackOption={noBackOption}
            onBack={NavigationService.back}
          />
          <H1 color="primary">
            {title}
          </H1>
          <Right
            noCloseOption={noCloseOption}
            onClose={onClose}
          />
        </View>
      </SafeAreaView>
    );
  }
}

export default StackHeader;

const styles = StyleSheet.create({
  wrapper: {
    // height: 45,
    // backgroundColor: '#fff',
    alignItems: 'flex-end',
    justifyContent: 'space-between',
    // flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: theme.primary[500],
    flexDirection: 'row'
  }
});
