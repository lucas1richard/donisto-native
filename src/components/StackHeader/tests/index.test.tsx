import * as React from 'react';
import { shallow } from 'enzyme';
import StackHeader from '..';

describe('<StackHeader />', () => {
  const renderComponent = () => shallow(<StackHeader />);
  it('renders', () => {
    expect(renderComponent()).toMatchSnapshot();
  });
});