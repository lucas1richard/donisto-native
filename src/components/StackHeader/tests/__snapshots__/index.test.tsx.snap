// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StackHeader /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StackHeader
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "alignItems": "flex-end",
            "borderBottomColor": "rgb(251, 100, 65)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Left
          onBack={[Function]}
        />
        <H1
          color="primary"
        >
          
        </H1>
        <Right />
      </Component>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Left
            onBack={[Function]}
          />,
          <H1
            color="primary"
          >
            
          </H1>,
          <Right />,
        ],
        "style": Object {
          "alignItems": "flex-end",
          "borderBottomColor": "rgb(251, 100, 65)",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "noBackOption": undefined,
            "onBack": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
            "color": "primary",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "noCloseOption": undefined,
            "onClose": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "alignItems": "flex-end",
              "borderBottomColor": "rgb(251, 100, 65)",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Left
            onBack={[Function]}
          />
          <H1
            color="primary"
          >
            
          </H1>
          <Right />
        </Component>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Left
              onBack={[Function]}
            />,
            <H1
              color="primary"
            >
              
            </H1>,
            <Right />,
          ],
          "style": Object {
            "alignItems": "flex-end",
            "borderBottomColor": "rgb(251, 100, 65)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "noBackOption": undefined,
              "onBack": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "color": "primary",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "noCloseOption": undefined,
              "onClose": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
